src/rosdistro/dependency_walker.py

```diff
@@ -93,8 +93,7 @@ def get_depends_on(self, pkg_name, depend_type, ignore_pkgs=None):
         for name in self._get_package_names():
             if name in ignore_pkgs:
                 continue
-            pkg = self._get_package(name)
-            repo = self._distribution_instance.repositories[pkg.repository_name].release_repository
```

Comment:
I think the `pkg.repository_name` logic should either be maintained, or the new function should be `_get_repo_for_package`, since right now it's a bit ambiguous whether it's "get the repo named X" or "get the repo for package named X".

Reply:
Valid point on the function name, I've changed it.

The reason that I changed how the release repo was determining the repo was b/c on my system it was continually failing with repository_name not a valid.  I dumped the pkg object and don't see repository_name being set.
I also checked in catkin_pkg parse_package_string and don't see it defined.  

If this is an issue with my version compatibility I can change it back but I'm unsure of how it's currently working for the normal DependencyWalker use case.

Reply:
It's set here: https://github.com/ros-infrastructure/rosdistro/blob/fc9b2441c5d79b918164c62a0f470f0934a6b172/src/rosdistro/package.py#L39

The `Package` object in question is from `catkin_pkg`; it's a structure internal to `rosdistro`. Anyway, changing the package name is totally satisfactory to me.